tests/test_models.py

```diff
@@ -98,19 +102,42 @@ def test_firebase_error(
         message: Message,
         mock_firebase_send: MagicMock,
         firebase_error: FirebaseError,
-        mock_fcm_device_deactivate: MagicMock,
     ):
         """
-        Ensure we call deactivate_devices_with_error_result and raise the FirebaseError
+        Ensure when happened unknown firebase error device is still active and raised the FirebaseError
         """
         mock_firebase_send.side_effect = firebase_error
 
         with pytest.raises(FirebaseError, match=str(firebase_error)):
             fcm_device.send_message(message)
 
-        mock_fcm_device_deactivate.assert_called_once_with(
-            fcm_device, fcm_device.registration_id, firebase_error
+        fcm_device.refresh_from_db()
+        # device is still active because error is unknown
+        assert fcm_device.active
+
+    def test_firebase_invalid_registration_error(
```

Comment:
Does it make sense to leave test writing to the future PR?

Reply:
It does. PRs should be about one thing. Otherwise they are much more expensive to polish, review and merge

Reply:
> It does. PRs should be about one thing. Otherwise they are much more expensive to polish, review and merge

Is your point that tests should be delivered after changes that those tests should check?

_There are three way to write tests:_
1. _Before writing the code_
2. _During writing the code_
3. _Never(some people call it "later")_